File: src/utils/logger.ts
// src/utils/consoleRedirect.ts
import { invoke } from "@tauri-apps/api/core";

// == Redirige les console.* vers le backend Rust ==
 
export function setupConsoleRedirect() {
  // correspondance exacte avec ton enum Rust
  const levels = ["Trace", "Debug", "Info", "Warn", "Error"] as const;

  // mapping entre console.* et ton enum
  const consoleMap: Record<string, typeof levels[number]> = {
    log: "Info",
    info: "Info",
    warn: "Warn",
    error: "Error",
    debug: "Debug",
  };

  for (const [method, rustLevel] of Object.entries(consoleMap)) {
    const original = (console as any)[method].bind(console);

    (console as any)[method] = (...args: any[]) => {
      const message = args.map(String).join(" ");
      invoke("frontend_log", { level: rustLevel, message }).catch(console.error);
      original(...args);
    };
  }

  console.info("{29.logger.ts} ✅ Console redirection initialized");
}

File: src-tauri/src/logger.rs
use serde::{Deserialize};
use tauri::{command};


#[derive(Deserialize)]
pub enum FrontendLogLevel {
    Trace,
    Debug,
    Info,
    Warn,
    Error,
}

#[command]
pub fn frontend_log(level: FrontendLogLevel, message: String) {
    match level {
        FrontendLogLevel::Trace => log::trace!(target: "frontend", "{}", message),
        FrontendLogLevel::Debug => log::debug!(target: "frontend", "{}", message),
        FrontendLogLevel::Info  => log::info!(target: "frontend", "{}", message),
        FrontendLogLevel::Warn  => log::warn!(target: "frontend", "{}", message),
        FrontendLogLevel::Error => log::error!(target: "frontend", "{}", message),
    }
}
